src/chocolatey/infrastructure.app/nuget/NugetCommon.cs

```diff
@@ -451,16 +451,21 @@ ISet<SourcePackageDependencyInfo> dependencyInfos
 
             //if (availablePackages.Contains(packageID)) return;
 
-            var dependencyInfoResources = resources.DependencyInfoResources();
-
-            foreach (var dependencyInfoResource in dependencyInfoResources)
+            foreach (var resource in resources)
             {
-                IEnumerable<SourcePackageDependencyInfo> dependencyInfos = Array.Empty<SourcePackageDependencyInfo>();
+                var dependencyInfoResource = resource.DependencyInfoResource;

```

Comment:
`DependencyInfoResources()` has a not null check, given that you are accessing the dependency info resource directly here, make sure to do check to make sure it is not null.

Reply:
I've updated it to match the null check in the GetPackagesDependencies overload that takes a PackageIdentity (matches lines 378-382).

Reply:
Did the null check added at line 458 address this?

Reply:
Based on what I can see, it looks like it is addressed. I am currently exploring the fix, as we had a number of concerns, and it looks very similar to a fix that AdmiringWorm tried to put in as a part of #3433, that didn't work for some reason.

I'll leave this conversation unresolved for now just to ensure that we get a full tests around this case.

Reply:
@josh-cooley, is there a reason that this isn't using the `DependencyInfoResources()` extension method? It feels like it's effectively doing the same thing the extension method does, just in more lines.

Reply:
A lot of the code is patterned after existing code in this class.  The structure of GetPackageDependencies that takes a PackageIdentity (line 365) is similar to this overload.  It did not use DependencyInfoResources(), so I matched the code there.

Reply:
> It did not use DependencyInfoResources(), so I matched the code there.

Looking at the file, it looks like each method was using a different way to get the DependencyInfoResources. As this was the code being editted, I think it makes sense to line them both up. I'm good with this here.